dove/tests/cmd_dove_ct_file.rs

```diff
@@ -0,0 +1,148 @@
+#![cfg(test)]
+
+use std::path::{Path, PathBuf};
+use std::fs::{remove_dir_all, read_to_string, remove_file};
+use fs_extra::file::write_all;
+use dove::cli::execute;
+
+/// $ dove ct -f ###
+/// project name: demoproject_20
+#[test]
+fn file() {
+    // Path to dove folder
+    let dove_folder = {
+        let mut folder = Path::new(".").canonicalize().unwrap();
+        if folder.to_str().unwrap().find("dove").is_none() {
+            folder.push("dove");
+        }
+        folder
+    };
+    // Project name and path
+    let project_name = "demoproject_20";
+    let project_folder = {
+        let mut folder = dove_folder.clone();
+        folder.push(project_name);
+        if folder.exists() {
+            remove_dir_all(&folder).expect(&format!(
+                "[ERROR] Couldn't delete project directory: {}",
+                folder.to_str().unwrap()
+            ));
+        }
+        folder
+    };
+
+    // $ cargo run -- new demoproject_20
+    // $ dove new demoproject_20
+    {
+        let args = &["dove", "new", project_name];
+        let command_string: String = args.join(" ").to_string();
+        execute(args, dove_folder.clone()).expect(&format!("[COMMAND] {}", &command_string));
+        set_dependencies_local_move_stdlib(&project_folder);
+    }
+
+    // $ cargo run -- build
+    // $ dove build
+    {
+        let args = &["dove", "build"];
+        let command_string: String = args.join(" ").to_string();
+        execute(args, project_folder.clone()).expect(&format!(
+            "[COMMAND] {}\r\n[FOLDER] {}",
+            &command_string,
+            project_folder.to_str().unwrap()
+        ));
+    }
+    // project_folder/scripts/sdemo_1.move
+    {
+        let mut path = project_folder.clone();
+        path.push("scripts");
+        path.push("sdemo_1.move");
+        write_all(
+            &path,
+            "script {
```

Comment:
nit: wouldn't it be possible to prepare such test files as regular files in some `test_assets` directory?

Reply:
They are all different and this is an important part of the test. Do I need to hide it?